lib/miner-info.js

```diff
@@ -29,6 +30,32 @@ async function getChainHead ({ maxAttempts = 5 } = {}) {
  * @returns {Promise<string>} Miner's PeerId, e.g. `12D3KooWMsPmAA65yHAHgbxgh7CPkEctJHZMeM3rAvoW8CZKxtpG`
  */
 export async function getMinerPeerId (minerId, { maxAttempts = 5 } = {}) {
+  // Make a concurrent request to both sources: RPC node and smart contract
+  const [rpcNodeResult, contractResult] = await Promise.allSettled([
+    getMinerPeerIdFromRpcNode(minerId, { maxAttempts }),
+    getMinerPeerIdFromSmartContract(minerId)
+  ])
+  // Check contract result first
+  if (contractResult.status === 'fulfilled' && contractResult.value) {
+    return contractResult.value
+  }
+
+  // Fall back to rpc-Node result
+  if (rpcNodeResult.status === 'fulfilled' && rpcNodeResult.value) {
+    return rpcNodeResult.value
+  }
+
+  // Handle the case where both failed
+  throw new Error(`Failed to resolve PeerID from any source: ${contractResult.reason}, ${rpcNodeResult.reason}`)
```

Comment:
Let's make the error message more descriptive, and format the error reasons in a way that will make them easier to read when reason is spanning multiple lines (e.g. an error stack trace).

```suggestion
  throw new Error(`Failed to obtain Miner's Index Provider PeerID.\nSmartContract query error: ${contractResult.reason}\nStateMinerInfo query error: ${rpcNodeResult.reason}`)
```